typed/remove.go

```diff
@@ -23,13 +23,15 @@ type removingWalker struct {
 	value    *value.Value
 	schema   *schema.Schema
 	toRemove *fieldpath.Set
+	shallow  bool
```

Comment:
Not entirely happy with the name `shallow`.
Thought about recursive, but didn't feel right as well.

Left it this way for now, happy for suggestions.